### PR TITLE
[BEAM-8108] Add Chicago Taxi Example running on Flink

### PR DESCRIPTION
**Pull request by kamilwu**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by kamilwu** [comment]
@kkucharc Could you trigger some tests for me, please?
First, I need a seed job to be run:
`Run Seed Job`

After that:
`Run Chicago Taxi on Dataflow`
`Run Chicago Taxi on Flink`

Thanks!

**Comment by kkucharc** [comment]
Run Seed Job

**Comment by aromanenko-dev** [comment]
Run Seed Job

**Comment by kkucharc** [comment]
Run Chicago Taxi on Dataflow

**Comment by aromanenko-dev** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
Thanks for your help @aromanenko-dev! I'm investigating the failure

**Comment by aromanenko-dev** [comment]
@kamilwu Please, let me know if you will need any other help.

**Comment by kkucharc** [comment]
Run Seed Job

**Comment by kkucharc** [comment]
Run Chicago Taxi on Dataflow

**Comment by kkucharc** [comment]
Run Chicago Taxi on Flink

**Comment by kkucharc** [comment]
retest this please

**Comment by kkucharc** [comment]
Run Seed Job

**Comment by kkucharc** [comment]
Run Chicago Taxi on Flink

**Comment by kkucharc** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
R: @pabloem Can I ask you for a review?

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+PostcommitJobBuilder.postCommitJob(
+    'beam_PostCommit_Python_Chicago_Taxi_Flink',
+    'Run Chicago Taxi on Flink',
+    'Google Cloud Flink Runner Chicago Taxi Example',
+    this
+) {
+    chicagoTaxiJob(delegate)
+}
+
+CronJobBuilder.cronJob(
```
Comment (Ardagan) [inline .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: PostcommitJobBuilder already adds periodic cron job
https://github.com/apache/beam/blob/5a24fc90a58dc1b05f2faf1ea1874d1fd044071a/.test-infra/jenkins/PostcommitJobBuilder.groovy#L59

Any reason to add extra one?

Comment (kamilwu) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: The cron job defined by PostcommitJobBuilder is executed every six hours, which is too frequent for our needs. The extra one overrides it with more desired setting (every 24 hours)

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import PostcommitJobBuilder
+
+def chicagoTaxiJob = { scope ->
+    scope.description('Runs the Chicago Taxi Example on the Flink runner.')
+    commonJobProperties.setTopLevelMainJobProperties(scope, 'master', 120)
+
+    def numberOfWorkers = 5
+
+    Docker publisher = new Docker(scope, LoadTestsBuilder.DOCKER_CONTAINER_REGISTRY)
+    publisher.publish(':sdks:python:container:py2:docker', 'python2.7_sdk')
```
Comment (pabloem) [inline .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: I wonder if we should be using a py3 container straight out

Comment (pabloem) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: though maybe then we'd have to run everything in py3 (meaning the target `:portable:py2:chicagoTaxiExample`). LMK what you think.

**Comment by pabloem** [comment]
feel free to merge afterwards @Ardagan @kkucharc

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy (hunk 2 above)
Comment (kamilwu) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: I ran the test in py3 (both SDK harness and the pipeline program) and there are some errors connected with `dill` package. What do you think about leaving it "as is" and creating a jira ticket to move chicago taxi to py3? We will have to do the migration anyway in the nearest future.

Comment (kamilwu) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Flink.groovy]: https://issues.apache.org/jira/browse/BEAM-9154

**Comment by kkucharc** [comment]
Run Seed Job

**Comment by kkucharc** [comment]
Run Chicago Taxi on Flink

**Comment by kamilwu** [comment]
Thanks everyone!

**Pull request merged**